### ISSUE TITLE
Write JSON asynchronously to stream with System.Text.Json

### ISSUE THREAD

**Issue by scharnyw** [dotnet/runtime#95194]
I have a requirement where I need to manually write JSON (i.e. at a level similar to the functionality provided by `Utf8JsonWriter`, instead of the `JsonSerializer` level) to a file stream. I want to do the writing to disk part asynchronously for performance reasons. I wonder if there's any way to achieve this easily with STJ?

I tried looking for async overloads in `Utf8JsonWriter` but couldn't find any. This is quite a surprise considering that 1. I assumed that `Utf8JsonWriter` is a low-level API on which higher level ones such as `JsonSerializer.SerializeAsync()` depended, and 2. writing JSON asynchronously is actually supported by `Newtonsoft.Json.JsonTextWriter` which is the Newtonsoft analogue of `Utf8JsonWriter`, and Newtonsoft predates STJ by many years and has much poorer support for async overall.

**Comment by huoyaoyuan** [comment]
Looking at the implementation of `Utf8JsonWriter`,

The write overloads only writes to its internal buffer:
https://github.com/dotnet/runtime/blob/cc4db9bd71f2dfcc29dc71385cd96916253c8850/src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.String.cs#L116-L140

And asynchronous write to stream is a separated operation:
https://github.com/dotnet/runtime/blob/cc4db9bd71f2dfcc29dc71385cd96916253c8850/src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs#L407-L447

In other words, it's low level so that you need to manage its buffering strategy. Calling asynchronous IO too frequently also hurts performance.

**Comment by scharnyw** [comment]
@huoyaoyuan That made sense after you explained it. Thanks a lot!
